pkg/connect/pubsub/proxy.go

```diff
@@ -0,0 +1,295 @@
+package pubsub
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	connect_sdk "github.com/inngest/inngestgo/connect"
+	"github.com/oklog/ulid/v2"
+	"github.com/redis/rueidis"
+	"sync"
+	"time"
+)
+
+type ProxyResponse struct {
+	Status string
+
+	SdkResponse *connect_sdk.SdkResponse
+}
+
+type RequestForwarder interface {
+	Proxy(ctx context.Context, data connect_sdk.GatewayMessageTypeExecutorRequestData) (*connect_sdk.SdkResponse, error)
+}
+
+type RequestReceiver interface {
+	ReceiveExecutorMessages(ctx context.Context, appId uuid.UUID, onMessage func(data connect_sdk.GatewayMessageTypeExecutorRequestData)) error
+	NotifyExecutor(ctx context.Context, appId uuid.UUID, resp ProxyResponse) error
+	AckMessage(ctx context.Context, appId uuid.UUID, requestId string) error
+
+	Wait(ctx context.Context) error
+}
+
+type redisPubSubConnector struct {
+	client       rueidis.Client
+	pubSubClient rueidis.DedicatedClient
+
+	subscribers     map[string]map[string]chan string
+	subscribersLock sync.RWMutex
+
+	RequestForwarder
+	RequestReceiver
+}
+
+func NewRedisPubSubConnector(client rueidis.Client) *redisPubSubConnector {
+	return &redisPubSubConnector{
+		client:          client,
+		subscribers:     make(map[string]map[string]chan string),
+		subscribersLock: sync.RWMutex{},
+	}
+}
+
+func (i *redisPubSubConnector) Proxy(ctx context.Context, data connect_sdk.GatewayMessageTypeExecutorRequestData) (*connect_sdk.SdkResponse, error) {
+	if data.RequestId == "" {
+		data.RequestId = ulid.MustNew(ulid.Now(), rand.Reader).String()
+	}
+
+	dataBytes, err := json.Marshal(data)
```

Comment:
Do we want to use a binary serializer instead of JSON?
WS allows binary transport so it can reduce the message sizes going back and forth.

Reply:
We should, the question is which one. Protobuf?

Reply:
unless there's a strong argument for something else, we can default to protobuf for now.